### PR TITLE
Update UI tests

### PR DESCRIPTION
**Pull request by pixlwave**
Follow up to #718 which makes the following changes:
- Update all failing snapshot tests in the codebase (some of these have been failing for quite a while it seems).
- Replace German snapshots with double-length pseudolanguage so that downloading new translations won't suddenly break our tests (also they seem much more extreme in a lot of cases too).
- Fix the logic for grouping messages with reactions.

Quite possibly easier to take this one commit-by-commit.

Run of the UI tests is going on [here](https://github.com/vector-im/element-x-ios/actions/runs/4542578034).

### CODE REVIEW

**Comment by manuroe** [comment]
>  double-length pseudolanguage 

TIL `NSDoubleLocalizedStrings`. Super idea 👍

File: ElementX/Sources/Services/Media/MockMediaProvider.swift [hunk 1]
```diff
@@ -23,14 +23,10 @@ struct MockMediaProvider: MediaProviderProtocol {
             return nil
         }
         
-        #warning("Fix me. this is stupid!")
```
Comment (stefanceriu) [inline ElementX/Sources/Services/Media/MockMediaProvider.swift:26]: Yey!

File: ElementX/Sources/Screens/RoomScreen/RoomScreenViewModel.swift [hunk 2]
```diff
@@ -196,7 +196,7 @@ class RoomScreenViewModel: RoomScreenViewModelType, RoomScreenViewModelProtocol
         }
         
         //  can be improved by adding a date threshold
-        return otherEventTimelineItem.properties.reactions.isEmpty && eventTimelineItem.sender == otherEventTimelineItem.sender
+        return eventTimelineItem.properties.reactions.isEmpty && eventTimelineItem.sender == otherEventTimelineItem.sender
```
Comment (stefanceriu) [inline ElementX/Sources/Screens/RoomScreen/RoomScreenViewModel.swift:199]: Even more yey, good catch!

File: UITests/Sources/Application.swift [hunk 3]
```diff
@@ -76,6 +76,11 @@ extension XCUIApplication {
         
         return name
     }
+    
+    private var localeCode: String {
+        guard !UserDefaults.standard.bool(forKey: "NSDoubleLocalizedStrings") else { return "pseudo" }
```
Comment (stefanceriu) [inline UITests/Sources/Application.swift:81]: This feels a bit convoluted, I would go with

```
if UserDefaults.standard.bool(forKey: "NSDoubleLocalizedStrings") == true {
   return "pseudo"
}
```

File: UnitTests/Sources/RoomScreenViewModelTests.swift [hunk 4]
```diff
@@ -14,6 +14,154 @@
 // limitations under the License.
 //
 
+@testable import ElementX
 import XCTest
 
-class RoomScreenViewModelTests: XCTestCase { }
+@MainActor
+class RoomScreenViewModelTests: XCTestCase {
```
Comment (stefanceriu) [inline UnitTests/Sources/RoomScreenViewModelTests.swift:21]: Awesome 👏

**Review by stefanceriu — APPROVED** [review]
Fabulous! 👏

**Pull request merged**